sdk/core/System.ClientModel/src/Pipeline/ClientLoggingPolicy.cs

```diff
@@ -0,0 +1,360 @@
+            }
+
+            _handler.LogRequestContent(requestId, bytes, requestTextEncoding);
+        }
+
+        var before = Stopwatch.GetTimestamp();
+
+        try
+        {
+            if (async)
+            {
+                await ProcessNextAsync(message, pipeline, currentIndex).ConfigureAwait(false);
+            }
+            else
+            {
+                ProcessNext(message, pipeline, currentIndex);
+            }
+        }
+        catch (Exception ex)
+        {
+            _handler.LogExceptionResponse(requestId, ex);
+
+            throw;
+        }
+
+        var after = Stopwatch.GetTimestamp();
+
+        PipelineResponse response = message.Response!;
+
+        double elapsed = (after - before) / (double)Stopwatch.Frequency;
+
+        if (response.IsError)
+        {
+            _handler.LogErrorResponse(requestId, response, elapsed);
+        }
+        else
+        {
+            _handler.LogResponse(requestId, response, elapsed);
+        }
+
+        if (_logContent && response.ContentStream != null && _handler.IsEnabled(LogLevel.Information, EventLevel.Informational))
+        {
+            Encoding? responseTextEncoding = null;
+
+            if (response.Headers.TryGetValue("Content-Type", out var contentType) && contentType != null)
+            {
+                ContentTypeUtilities.TryGetTextEncoding(contentType, out responseTextEncoding);
+            }
+
+            if (message.BufferResponse || response.ContentStream.CanSeek)
+            {
+                byte[]? responseBytes;
+                if (message.BufferResponse)
+                {
+                    // Content is buffered, so log the first _maxLength bytes
+                    ReadOnlyMemory<byte> contentAsMemory = response.Content.ToMemory();
+                    var length = Math.Min(contentAsMemory.Length, _maxLength);
+                    responseBytes = contentAsMemory.Span.Slice(0, length).ToArray();
+                }
+                else
+                {
+                    responseBytes = new byte[_maxLength];
+                    response.ContentStream.Read(responseBytes, 0, _maxLength);
+                    response.ContentStream.Seek(0, SeekOrigin.Begin);
+                }
+
+                if (response.IsError)
+                {
+                    _handler.LogErrorResponseContent(requestId, responseBytes, responseTextEncoding);
+                }
+                else
+                {
+                    _handler.LogResponseContent(requestId, responseBytes, responseTextEncoding);
+                }
+            }
+            else
+            {
+                response.ContentStream = new LoggingStream(_handler, requestId, _maxLength, response.ContentStream, response.IsError, responseTextEncoding);
+            }
+        }
+
+        if (elapsed > RequestTooLongSeconds)
+        {
+            _handler.LogResponseDelay(requestId, elapsed);
```

Comment:
Should this log event also go in the transport intead?

Reply:
Yeah this one could probably go in the transport since it's a warning.